src/sw/Core/Src/EventManager.cpp

```diff
@@ -0,0 +1,19 @@
+#include "EventManager.h"
+
+void EventManager::subscribe(TemperatureSensor observers)
+{
+	m_subscribers.push_back(observers);
+}
+void EventManager::unsubscribe(TemperatureSensor observers)
+{
+
+}
+
+void EventManager::notify()
+{
+	temperatures.clear();
+	for (int i = 0; i < m_subscribers.size(); i++)
+	{
+		temperatures.push_back(m_subscribers[i].update());
```

Comment:
so the sensor is he subscriber? Should not be. The sensor is a vlog star who is subscribed by many 😉